hydra/utils.py

```diff
@@ -43,6 +43,11 @@ def get_static_method(full_method_name):
 
 def instantiate(config, *args, **kwargs):
     assert config is not None, "Input config is None"
+    assert (
+        isinstance(config.params, (DictConfig, ListConfig)) or config.params is None
```

Comment:
@omry The reason why I'm keeping ListConfig is that OmegaConf handles that by raising

    Merging DictConfig with ListConfig is not supported

The exception says "not supported" and doesn't throw a semantic error. If later on OmegaConf decides to add that support, we would still be limiting that in the Hydra layer even though it would work.

Reply:
So if you removed ListConfig here, wouldn't this code throw here instead which is the desired result?

Reply:
Nope; I _think_ that we should leave `ListConfig` handling to `OmegaConf`.

Reply:
I suggest that you just check that the params is None or DictConfig.
Not seeing an extra value by allowing ListConfig here.